src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -113,9 +124,52 @@ case class Cluster(clusterName: ClusterName,
   def projectNameString: String = s"${googleProject.string}/${clusterName.string}"
 }
 
+object MachineConfig {
+  implicit val machineConfigSemigroup = new Semigroup[MachineConfig] {
+    def combine(defined: MachineConfig, default: MachineConfig): MachineConfig = {
+      defined.numberOfWorkers match {
+        case None | Some(0) => MachineConfig(Some(0), defined.masterMachineType.orElse(default.masterMachineType), defined.masterDiskSize.orElse(default.masterDiskSize))
+        case Some(numWorkers) if numWorkers < 2 => throw TooFewWorkersRequestedException(numWorkers)
+        case _ => MachineConfig(defined.numberOfWorkers.orElse(defined.numberOfWorkers),
+                                defined.masterMachineType.orElse(default.masterMachineType),
+                                defined.masterDiskSize.orElse(default.masterDiskSize),
+                                defined.workerMachineType.orElse(default.workerMachineType),
+                                defined.workerDiskSize.orElse(default.workerDiskSize),
+                                defined.numberOfWorkerLocalSSDs.orElse(default.numberOfWorkerLocalSSDs),
+                                defined.numberOfPreemptibleWorkers.orElse(default.numberOfPreemptibleWorkers))
+      }
+    }
+  }
+
+  def apply(definedMachineConfig: Option[MachineConfig], defaultMachineConfig: ClusterDefaultsConfig): MachineConfig = {
+    definedMachineConfig.getOrElse(MachineConfig()) |+| MachineConfig(defaultMachineConfig)
```

Comment:
I think this could just be:
```
definedMachineConfig |+| MachineConfig(defaultMachineConfig)
```

If the LHS is `None` then it should use the RHS.

Reply:
`definedMachineConfig |+| MachineConfig(defaultMachineConfig)` gives me a `Type mistmatch` :( 
I suppose because you can't do `|+|` on `Option[MachineConfig]` and `MachineConfig`

Reply:
Oh, duh. Nevermind